src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -12,16 +13,41 @@ namespace Microsoft.Extensions.Logging.Console
     [UnsupportedOSPlatform("browser")]
     internal class ConsoleLoggerProcessor : IDisposable
     {
-        private const int _maxQueuedMessages = 1024;
+        private static Meter _meter = new Meter("Microsoft-Extension-Logging-Console-Queue", "1.0.0");
+        private readonly Queue<LogMessageEntry> _messageQueue;
+        private int _messagesDropped;
+        private bool _isAddingCompleted;
+        private int _maxQueuedMessages = 1048576;
+        public int MaxQueueLength
+        {
+            get => _maxQueuedMessages;
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentException(nameof(MaxQueueLength));
```

Comment:
If you used ArgumentOutOfRangeException you could use the constructor that accepts the value that was invalid, and also a message that explains what the valid range is.